index.js

```diff
@@ -135,31 +139,31 @@ class Plugin extends PluginMiniAccounts {
   }
 
   _handleIncomingPrepare (account, ilpData) {
-    const { amount } = IlpPacket.deserializeIlpPrepare(ilpData)
+    const amount = new EthUnit(
+      IlpPacket.deserializeIlpPrepare(ilpData).amount,
+      'gwei'
+    )
 
     const secured = account.getSecuredBalance()
     const prepared = account.getBalance()
     const newPrepared = prepared.add(amount)
     const unsecured = newPrepared.sub(secured)
-    debug(unsecured.toString(), 'unsecured; secured balance is',
-      secured.toString(), 'prepared amount', amount, 'newPrepared',
-      newPrepared.toString(), 'prepared', prepared.toString())
 
-    if (unsecured.greaterThan(this._bandwidth)) {
-      throw new Error('Insufficient bandwidth, used: ' + unsecured + ' max: ' +
```

Comment:
We should add a maxPacketAmount like in ilp-plugin-xrp-asym server. It's important for supporting ILP protocol STREAM

Reply:
Reading through xrp, this makes a lot more sense now. (As it turns out, `_handleIncomingPrepare` was never even called to begin with!). As a side note, would there be a way to extend ilp-plugin-mini-balances to abstract away the balance logic (somewhat)? Looks like it'd need to be more extensible in handling custom BTP messages, but just curious where you envisioned that plugin being used or if it's just better for each server plugin to roll its own balance logic like xrp-asym-server.

Reply:
ILP Plugin Payment (https://github.com/interledgerjs/ilp-plugin-payment) is pretty much whatt you're asking for. It's still rough around the edges but should be usable once https://github.com/interledgerjs/ilp-plugin-payment/pull/1 this is merged. It includes an example of how it could be used to create an XRP payment based plugin, so you could start off of that

Reply:
Thanks. Using ilp-plugin-payment for an on-ledger integration makes sense to me, but I'm a little confused in this case though, how would the paychan updates be exchanged between client and server with it?